### DISCUSSION TITLE
NerDLAproach bestModelMetric "Param bestModelMetric does not exist" Error

### DISCUSSION THREAD

**Discussion by zetobur** [JohnSnowLabs/spark-nlp#7469] [Q&A] [answered]
I encountered an error which says "java.util.NoSuchElementException: Param bestModelMetric does not exist." when I try to execute [https://nlp.johnsnowlabs.com/docs/en/serving_spark_nlp_via_api_databricks_mlflow](url) example on Databricks.

Then I saw that this metric is newly committed and saying "[Make sure bestModelMetric is being used](https://github.com/JohnSnowLabs/spark-nlp/commit/043f0e9b6cb07c5fc4fcf165043bb22f2969187d)". But at first I couldn't find what the value should be. Then found some explanations (#6749 ) about this and inside them there are values such as "F1 Micro-average, F1 Macro-average, f1_micro, f1_macro,  microF1, macrof1".

But whatever I tried all of them one by one inside setBestModelMetric, and setUseBestModel(True) still getting same error. 

Here is the code:
`
MODEL_NAME = 'nlp_mlflow'
EXPERIMENT_NAME = 'test_4'
OUTPUT_DIR = f"{MODEL_NAME}_{EXPERIMENT_NAME}_output" # Output folder of all your model artifacts
MODEL_DIR = f"/Users/sgurleye@its.jnj.com/" # Name of the folder where the MLFlow model will be stored

MAX_EPOCHS = 2
LEARNING_RATE = 0.001
BATCH_SIZE = 8
RANDOM_SEED = 0
VALIDATION_SPLIT = 0.2
BESTMODELMETRIC = "microf1"

document = DocumentAssembler()\
        .setInputCol("text")\
        .setOutputCol("document")

sentence = SentenceDetector()\
        .setInputCols(['document'])\
        .setOutputCol('sentence')

token = Tokenizer()\
        .setInputCols(['sentence'])\
        .setOutputCol('token')

embeddings = BertEmbeddings.pretrained("bert_base_cased", "en") \
      .setInputCols("sentence", "token") \
      .setOutputCol("embeddings")

ner_approach = NerDLApproach()\
      .setInputCols(["sentence", "token", "embeddings"])\
      .setLabelColumn("label")\
      .setOutputCol("ner")\
      .setMaxEpochs(MAX_EPOCHS)\
      .setLr(LEARNING_RATE)\
      .setBatchSize(BATCH_SIZE)\
      .setRandomSeed(RANDOM_SEED)\
      .setVerbose(1)\
      .setEnableOutputLogs(True)\
      .setIncludeConfidence(True)\
      .setIncludeAllConfidenceScores(True)\
      .setEvaluationLogExtended(True)\
      .setOutputLogsPath(OUTPUT_DIR)\
      .setValidationSplit(VALIDATION_SPLIT)\
      .setUseBestModel(True)\
      .setBestModelMetric(BESTMODELMETRIC)

ner_preprocessing_pipeline = Pipeline(stages=[
    document,
    sentence,
    token,
    embeddings
 ])

ner_training_pipeline = Pipeline(stages = ner_preprocessing_pipeline.getStages() + [ner_approach])

ner_model = ner_training_pipeline.fit(training_data)
`

Still receiving the same error "java.util.NoSuchElementException: Param bestModelMetric does not exist."

Could you clarify me please? Is there anything wrong in the example? @maziyarpanahi

**Comment by maziyarpanahi** [comment, marked as answer]
Answered here https://github.com/JohnSnowLabs/spark-nlp/issues/7470
